.github/workflows/trigger-deployments.yml

```diff
@@ -0,0 +1,53 @@
+name: Trigger deployment on test then prod
+
+on:
+  push:
+    tags:
+    - '*'
+
+jobs:
+  trigger-test-deployment:
+    runs-on: ubuntu-latest
+    environment: test
+    steps:
+    - run: echo ${{ github.ref_name }} > ~/tag_to_deploy
+    - run: chmod 660 ~/tag_to_deploy
+    - run: touch ~/ssh_id
+    - run: chmod 700 ~/ssh_id
+    - run: echo "${{ secrets.BUILD_SSH_PRIVATE_KEY }}" > ~/ssh_id
+    - run: echo "${{ secrets.BUILD_SSH_PUBLIC_KEY }}" > ~/ssh_id.pub
+    - run: touch ~/my_known_hosts
+    - run: chmod 700 ~/my_known_hosts
+    - run: echo "${{ secrets.KNOWN_HOSTS }}" > ~/my_known_hosts
+    - run: scp -v -i ~/ssh_id -o UserKnownHostsFile=~/my_known_hosts -p ~/tag_to_deploy build@${{ secrets.MACHINE_ADDRESS }}:/home/build/deployment/tag_to_deploy
+    - uses: artiz/poll-endpoint@1.0.2
+      with:
+        url: ${{ secrets.DEPLOYED_VERSION_URL }}
+        method: GET
+        expectStatus: 200
+        expectBody: "${{ github.ref_name }}"
+        timeout: 600000
+        interval: 20000
+  trigger-production-deployment:
```

Comment:
This looks identical to `trigger-test-deployment` -- do they both push to the same place?  What's the point of having both?  Is there a way to DRY this at all?

Reply:
@slifty You are right that the body is identical and it would be better to make it DRY. They are identical except for the environment. The environment substitutes the appropriate variables. I find [evidence that we can break it out](https://docs.github.com/en/actions/using-workflows/reusing-workflows). I think the only downside is this will mean another `.yml` file separate from this one. But it is probably worth it because yes, it's exactly the same code with different environment variable values.

Reply:
@slifty And no, they do not push to the same place, one pushes to the test environment (think "deploy preview") and the other pushes to the production environment.

Reply:
@slifty The point of having both is to deploy to the test environment prior to deploying to the production environment. If everything works during test deployment such that the services are up and responsive, this increases confidence and reduces the risk of something going wrong when deploying to production.

Reply:
Ahh got it I didn't realize the environment dictated that -- makes sense!

DRY would be nice to have, but I defer to you on what makes sense.  Thanks for the explanation!

Reply:
OK, I made it DRYer than before by putting the script steps into an actual bash script. This still causes some indirection but it should be fairly easy to read and follow.

Some benefits include
* the script is now in the repo,
* documented, and
* `shellcheck`ed when changed (#80 #103).

The downside is the one step of indirection, the uncertainty that this will actually work when pushed (but we can fix it), and there are still a couple of steps repeated (check out and poll the endpoint).

What do you think?